frontend/src/utilities/dataEntryToRecord.ts

```diff
@@ -0,0 +1,12 @@
+type GenericEnvVariableDataEntry<T> = {
+  key: T;
+  value: string;
+}[];
+
+export const dataEntryToRecord = <T extends string>(
+  dataRecord: GenericEnvVariableDataEntry<T>,
+): Partial<Record<T, string>> => {
+  const initialRecord: Partial<Record<T, string>> = {};
+
+  return dataRecord.reduce((acc, { key, value }) => ({ ...acc, [key]: value }), initialRecord);
```

Comment:
```suggestion
  return dataRecord.reduce<...>((acc, { key, value }) => ({ ...acc, [key]: value }), initialRecord);
```
I think you can just use the generic as part of reduce